### PR TITLE
Fixed #26511 - Added documentation for substring search in JSONField

### PR DESCRIPTION
**Pull request by jensneuhaus**

### CODE REVIEW

File: docs/ref/contrib/postgres/fields.txt [hunk 1]
```diff
@@ -358,6 +358,19 @@ field. It uses the SQL operator ``@>``. For example::
...
+If you want to search for objects where the title contains a substring instead of an
+exact match, you'll have to cast your JSON field to text with :class:`~django.db.models.functions.Cast`
+before hand. For example::
+
+    >>> Dog.objects.create(name='Rufus', data={'hobby': 'Hanging around with cats'})
+
+    >>> from django.db import models
+    >>> from django.db.models.functions import Cast
+    >>> Dog.objects.annotate(
+    ...     text=Cast('data', models.TextField())
```
Comment (charettes) [inline docs/ref/contrib/postgres/fields.txt]: Maybe we could mention how to search a key instead of the whole JSON blob.

Comment (jensneuhaus) [inline docs/ref/contrib/postgres/fields.txt]: I tried Cast('data__hobby'...) like you suggested in the ticket, but it did not work. I would get the following error message:

"FieldError: Cannot resolve keyword 'hobby' into field. Join on 'data' not permitted."

Comment (timgraham) [inline docs/ref/contrib/postgres/fields.txt]: Simon, do you want to proceed with this patch as-is and file a ticket for that enhancement?

Comment (charettes) [inline docs/ref/contrib/postgres/fields.txt]: Searching a whole JSON blob seems like a bad practice to me.

I'm surprised `Cast('data__hobby', models.TextField())` isn't working. @mjtamlyn is there a way to use `KeyTransform` here?

Comment (charettes) [inline docs/ref/contrib/postgres/fields.txt]: The only expression I could come with is `Cast(models.Func('data', template="%(expressions)s -> 'bar'"), models.TextField())`

Comment (charettes) [inline docs/ref/contrib/postgres/fields.txt]: Actually `models.functions.Cast(KeyTransform('bar', 'data'), models.TextField())` kinda works but `KeyTransform` is not part of the public API and `Cast` needs adjustment on PostgreSQL.

Comment (charettes) [inline docs/ref/contrib/postgres/fields.txt]: Based on further investigation it looks like what we really need here is support for the `->>` operator as `(lhs -> key)::text` is the JSON representation of the string (`('"foo"'::jsonb)::text` -> `'"foo"'`).

I believe this should be done by introducing a new expression for this purpose and documenting it. In the meantime we could document using `Func('data', template="%(expressions)s ->> 'hobby'", output_field=models.TextField())`.

**Comment by timgraham** [comment]
I guess we'll close this for now.

**Pull request closed without merging**